### DISCUSSION TITLE
Integrating Azure AI Language Service into Presidio Analyzer

### DISCUSSION THREAD

**Discussion by whalygood** [microsoft/presidio#1456] [Q&A]
Hello Microsoft Presidio Team,

I am part of the Microsoft Startup Hub program and working with members of the Azure Advisory Team to integrate the Azure AI Language service for PII sanitization using the Microsoft Presidio containers. The Azure team suggested that I reach out to you.

I have an AI application and middleware proxy which makes a call to the Presidio Analyzer and Anonymizer containers with the users' prompt, and expects to receive a callback with the PII (e.g. names replaced by <PERSON>) deleted.

The default SpaCY NLP model misses some less common names, so I want to integrate the commercial AAI service as [described here](https://microsoft.github.io/presidio/samples/python/text_analytics/). In my Docker deploy-compose.override.yml file I have the following services defined:

```
  presidio-anonymizer:
    image: http://mcr.microsoft.com/presidio-anonymizer
    restart: always
    ports:
      - "127.0.0.1:5001:3000"
  presidio-analyzer:
    image: http://mcr.microsoft.com/presidio-analyzer
    restart: always
    ports:
      - "127.0.0.1:5002:3000"
    environment:
      - AZURE_AI_KEY=REDACTED
      - AZURE_AI_ENDPOINT=https://REDACTED.azure.com/
    volumes:
      - ./presidio/default_recognizers.yaml:/usr/bin/presidio-analyzer/presidio_analyzer/conf/default_recognizers.yaml
```

I am attempting to use the predefined recognizer at https://github.com/microsoft/presidio/blob/main/presidio-analyzer/presidio_analyzer/predefined_recognizers/azure_ai_language.py by adding the following lines to the default_recognizers.yaml file which I am bind mounting into the Analyzer container with the following lines added:

```
  - name: AzureAILanguageRecognizer
    type: predefined

```
Unfortunately when I attempt to start up the Analyzer container, it crash loops with this error in the container logs:

```
Traceback (most recent call last):
  File "/usr/bin/presidio-analyzer/app.py", line 138, in <module>
    server = Server()
  File "/usr/bin/presidio-analyzer/app.py", line 44, in __init__
    self.engine: AnalyzerEngine = AnalyzerEngineProvider(
  File "/usr/bin/presidio-analyzer/presidio_analyzer/analyzer_engine_provider.py", line 75, in create_engine
    registry = self._load_recognizer_registry(
  File "/usr/bin/presidio-analyzer/presidio_analyzer/analyzer_engine_provider.py", line 121, in _load_recognizer_registry
    registry = provider.create_recognizer_registry()
  File "/usr/bin/presidio-analyzer/presidio_analyzer/recognizer_registry/recognizer_registry_provider.py", line 59, in create_recognizer_registry
    recognizers = RecognizerListLoader.get(
  File "/usr/bin/presidio-analyzer/presidio_analyzer/recognizer_registry/recognizers_loader_utils.py", line 229, in get
    recognizer_instances.append(recognizer_cls(**kwargs))
TypeError: __init__() got an unexpected keyword argument 'context'
```

Would you be able to provide any guidance what I can do in order to get the integration working? It's not totally clear to me from the documentation how to add the AAI service to the Recognizer Registry – what additionally I need to do.

Thanks in advance.

**Comment by omri374** [comment]
Hi @whalygood, there's indeed a bug here. The `AzureAILanguageRecognizer` should include a `context` parameter as input (even if it's not using it). I'll create a PR shortly.
